File: nodes/timestamp/timestamp.process.py
from datetime import datetime
from typing import Any, Dict, List, Union

async def process(inputs: Dict[str, Any], settings: Dict[str, Any], config: Dict[str, Any], nodeConfig: Dict[str, Any]) -> Dict[str, Any]:
    try:
        format_type = inputs.get('format', 'iso')
        
        now = datetime.now()
        unix_timestamp = int(now.timestamp())
        unix_timestamp_ms = int(now.timestamp() * 1000)
        iso_string = now.isoformat() + 'Z'
        
        # Check for preset format keywords first
        if format_type == 'iso':
            timestamp = iso_string
        elif format_type == 'unix':
            timestamp = str(unix_timestamp)
        elif format_type == 'unix_ms':
            timestamp = str(unix_timestamp_ms)
        elif format_type == 'rfc2822':
            timestamp = now.strftime('%a, %d %b %Y %H:%M:%S GMT')
        else:
            # If not a preset keyword, treat as custom format string
            timestamp = format_custom_timestamp(now, format_type)
        
        return {
            'timestamp': timestamp,
            'unix_timestamp': unix_timestamp,
            'unix_timestamp_ms': unix_timestamp_ms,
            'iso_string': iso_string
        }
        
    except Exception as error:
        raise Exception(f"Timestamp error: {str(error)}")

def format_custom_timestamp(date, format_str):
    """Helper function for custom format formatting"""
    year = date.year
    month = f"{date.month:02d}"
    day = f"{date.day:02d}"
    hours = f"{date.hour:02d}"
    minutes = f"{date.minute:02d}"
    seconds = f"{date.second:02d}"
    milliseconds = f"{date.microsecond // 1000:03d}"
    
    return (format_str
            .replace('YYYY', str(year))
            .replace('MM', month)
            .replace('DD', day)
            .replace('HH', hours)
            .replace('mm', minutes)
            .replace('ss', seconds)
            .replace('SSS', milliseconds)
            .replace('YY', str(year)[-2:])
            .replace('M', str(date.month))
            .replace('D', str(date.day))
            .replace('H', str(date.hour))
            .replace('m', str(date.minute))
            .replace('s', str(date.second)))

File: nodes/timestamp/timestamp.process.js
export default async ({inputs, settings, config, nodeConfig}) => {
  try {
    const format = inputs.format || 'iso';
    
    const now = new Date();
    const unixTimestamp = Math.floor(now.getTime() / 1000);
    const unixTimestampMs = now.getTime();
    const isoString = now.toISOString();
    
    let timestamp;
    
    // Check for preset format keywords first
    switch (format) {
      case 'iso':
        timestamp = isoString;
        break;
        
      case 'unix':
        timestamp = unixTimestamp.toString();
        break;
        
      case 'unix_ms':
        timestamp = unixTimestampMs.toString();
        break;
        
      case 'rfc2822':
        timestamp = now.toUTCString();
        break;
        
      default:
        // If not a preset keyword, treat as custom format string
        timestamp = formatCustomTimestamp(now, format);
    }
    
    return {
      timestamp: timestamp,
      unix_timestamp: unixTimestamp,
      unix_timestamp_ms: unixTimestampMs,
      iso_string: isoString
    };
    
  } catch (error) {
    throw new Error(`Timestamp error: ${error.message}`);
  }
};

// Helper function for custom format formatting
function formatCustomTimestamp(date, format) {
  const year = date.getFullYear();
  const month = String(date.getMonth() + 1).padStart(2, '0');
  const day = String(date.getDate()).padStart(2, '0');
  const hours = String(date.getHours()).padStart(2, '0');
  const minutes = String(date.getMinutes()).padStart(2, '0');
  const seconds = String(date.getSeconds()).padStart(2, '0');
  const milliseconds = String(date.getMilliseconds()).padStart(3, '0');
  
  return format
    .replace(/YYYY/g, year)
    .replace(/MM/g, month)
    .replace(/DD/g, day)
    .replace(/HH/g, hours)
    .replace(/mm/g, minutes)
    .replace(/ss/g, seconds)
    .replace(/SSS/g, milliseconds)
    .replace(/YY/g, String(year).slice(-2))
    .replace(/M/g, String(date.getMonth() + 1))
    .replace(/D/g, String(date.getDate()))
    .replace(/H/g, String(date.getHours()))
    .replace(/m/g, String(date.getMinutes()))
    .replace(/s/g, String(date.getSeconds()));
}
